### PR TITLE
Tested book functionality

### PR DESCRIPTION
**Pull request by ivan0dyatlyukkk**
image
image

### CODE REVIEW

**Review by ConfusedSatlan — APPROVED** [review]
well done

File: src/test/java/org/diatliuk/bookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Verify the create() method by using valid request dto data")
+    @Sql(
+            scripts = DELETE_TEST_BOOK,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    void create_withValidCreateBookRequest_returnsBook() throws Exception {
+        BookDto expectedBook = new BookDto()
+                .setId(1L)
```
Comment (andrii-hoienko) [inline src/test/java/org/diatliuk/bookstore/controller/BookControllerTest.java:112]: What is the reason to populate expectedBook with a lot of fields if only id is checked?

File: src/test/java/org/diatliuk/bookstore/service/BookServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        assertEquals(RESPONSE_BOOK_DTO, actualBook);
+    }
+
+    @Test
+    @DisplayName("Verify getById() method by using a valid id")
+    void getById_withInvalidId_throwsException() {
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        assertThrows(EntityNotFoundException.class, () -> bookService.getById(INVALID_BOOK_ID));
```
Comment (andrii-hoienko) [inline src/test/java/org/diatliuk/bookstore/service/BookServiceTest.java]: Please check error message as well

File: src/test/resources/database/queries/books/delete-books-and-categories.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM books_categories;
+
+DELETE FROM books;
+
+DELETE FROM categories;
```
Comment (andrii-hoienko) [inline src/test/resources/database/queries/books/delete-books-and-categories.sql]: dont forget about empty lines

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ! Let`s improve your solution

File: src/test/resources/database/queries/books/delete-books-and-categories.sql (hunk 3 above)
Comment (ivan0dyatlyukkk) [reply src/test/resources/database/queries/books/delete-books-and-categories.sql]: My bad! I didn't spot it(

File: src/test/java/org/diatliuk/bookstore/controller/BookControllerTest.java (hunk 1 above)
Comment (ivan0dyatlyukkk) [reply src/test/java/org/diatliuk/bookstore/controller/BookControllerTest.java:112]: Sorry, I didn't catch a bit.
Did you mean this case?
assertTrue(EqualsBuilder.reflectionEquals(expectedBook, actualBook, "id"));

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**